third_party/2and3/werkzeug/http.pyi

```diff
@@ -1,36 +1,89 @@
+)
 
-def wsgi_to_bytes(data): ...
-def bytes_to_wsgi(data): ...
-def quote_header_value(value, extra_chars='', allow_token=True): ...
-def unquote_header_value(value, is_filename=False): ...
-def dump_options_header(header, options): ...
-def dump_header(iterable, allow_token=True): ...
-def parse_list_header(value): ...
-def parse_dict_header(value, cls=...): ...
-def parse_options_header(value, multiple=False): ...
-def parse_accept_header(value, cls=None): ...
-def parse_cache_control_header(value, on_update=None, cls=None): ...
-def parse_set_header(value, on_update=None): ...
-def parse_authorization_header(value): ...
-def parse_www_authenticate_header(value, on_update=None): ...
-def parse_if_range_header(value): ...
-def parse_range_header(value, make_inclusive=True): ...
-def parse_content_range_header(value, on_update=None): ...
-def quote_etag(etag, weak=False): ...
-def unquote_etag(etag): ...
-def parse_etags(value): ...
-def generate_etag(data): ...
-def parse_date(value): ...
-def cookie_date(expires=None): ...
-def http_date(timestamp=None): ...
-def is_resource_modified(environ, etag=None, data=None, last_modified=None, ignore_if_range=True): ...
-def remove_entity_headers(headers, allowed=...): ...
-def remove_hop_by_hop_headers(headers): ...
-def is_entity_header(header): ...
-def is_hop_by_hop_header(header): ...
-def parse_cookie(header, charset='', errors='', cls=None): ...
-def dump_cookie(key, value='', max_age=None, expires=None, path='', domain=None, secure=False, httponly=False, charset='', sync_expires=True): ...
-def is_byte_range_valid(start, stop, length): ...
+if sys.version_info < (3,):
+    _Str = TypeVar('_Str', str, unicode)
+    _ToBytes = Union[bytes, bytearray, buffer, unicode]
+    _ETagData = Union[str, unicode, bytearray, buffer, memoryview]
+else:
+    _Str = str
+    _ToBytes = Union[bytes, bytearray, memoryview, str]
+    _ETagData = Union[bytes, bytearray, memoryview]
+
+_T = TypeVar("_T")
+_U = TypeVar("_U")
+
+HTTP_STATUS_CODES: Dict[int, str]
+
+def wsgi_to_bytes(data: Union[bytes, Text]) -> bytes: ...
+def bytes_to_wsgi(data: bytes) -> str: ...
+def quote_header_value(value: Any, extra_chars: str = ..., allow_token: bool = ...) -> str: ...
+def unquote_header_value(value: _Str, is_filename: bool = ...) -> _Str: ...
+def dump_options_header(header: Optional[_Str], options: Mapping[_Str, Any]) -> _Str: ...
+def dump_header(iterable: Union[Iterable[Any], Dict[_Str, Any]], allow_token: bool = ...) -> _Str: ...
+def parse_list_header(value: _Str) -> List[_Str]: ...
+@overload
+def parse_dict_header(value: Union[bytes, Text]) -> Dict[Text, Optional[Text]]: ...
+@overload
+def parse_dict_header(value: Union[bytes, Text], cls: Type[_T]) -> _T: ...
+@overload
+def parse_options_header(value: None, multiple: bool = ...) -> Tuple[str, Dict[str, Optional[str]]]: ...
+@overload
+def parse_options_header(value: Optional[_Str], multiple: bool = ...) -> Tuple[_Str, Dict[_Str, Optional[_Str]]]: ...
```

Comment:
It looks it can actually return a tuple of more than two values, though only if multiple is True, unless I'm misreading the code.

Reply:
Oh, indeed, it actually says so in the docstring. I don't think there is a better way to express this than to use `Tuple[Any, ...]` (except for an overload when multiple is not given), is there?

Reply:
No, I don't think so.